kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/challenges/flightsearch/FlightSearchEngine.java

```diff
@@ -0,0 +1,53 @@
+package com.kodilla.good.patterns.challenges.flightsearch;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class FlightSearchEngine {
+
+    private Set<Flight> listOfFlights;
+
+    public FlightSearchEngine(Set<Flight> listOfFlights) {
+        this.listOfFlights = listOfFlights;
+    }
+
+    public Set<Flight> searchByFlightDeparture(String departure) {
+        System.out.println("Searching all Flights by departure from: " + departure);
+        listOfFlights.stream()
+                .filter(dep -> dep.getDeparture().equals(departure))
+                .forEach(System.out::println);
+        return listOfFlights.stream()
+                .filter(dep -> dep.getArrival().equals(departure))
+                .collect(Collectors.toSet());
+    }
+
+    public Set<Flight> searchByFlightArrival(String arrival) {
+        System.out.println("Searching all Flights by arrival to: " + arrival);
+        listOfFlights.stream()
+                .filter(arr -> arr.getArrival().equals(arrival))
+                .map(string -> string.toString())
+                .forEach(System.out::println);
+        return listOfFlights.stream()
+                .filter(arr -> arr.getArrival().equals(arrival))
+                .collect(Collectors.toSet());
+    }
+
+    public void searchByFlightConnection(String departure, String arrival, String through) {
```

Comment:
Zobacz ciekawą rzecz. Powyższe dwie metody zwracają `Set<Flight>`, a tutaj już jest `void`, ponieważ klasa `Flight` jest zbyt uboga, aby wyrazić lot z przesiadką.
Powinien to być dla Ciebie sygnał, że potrzebujesz dwóch różnych klas:
- pierwsza, określająca pojedynczy lot w siatce połączeń,
- druga, określająca wynik wyszukiwania (czyli możliwość uwzględnienia lotu z przesiadką). Może być ona stworzona na różne sposoby, np. departure/through/arrival (tak jak masz tutaj), albo w bardziej uniwersalny sposób jako lista pojedynczych połączeń - to pozwoli Ci wyrażać loty z więcej niż 1 przesiadką.